### ISSUE TITLE
[Linux D3D9] Guitar Hero World Tour Definitive Edition softlocks in song or when finding players

### ISSUE THREAD

**Issue by ZedekThePlagueDoctor** [doitsujin/dxvk#3037]
Guitar Hero World Tour with Definitive Edition mod softlocks at seemingly random points, sometimes when finding players in online play, other times during a song. The attached log is after a softlock in online play. This issue does not occur using WineD3D.

This using a 32-bit DXVK DLL referenced in issue #3000, obtained as an artifact from: https://github.com/doitsujin/dxvk/actions/runs/3268994708.

### Software information
Guitar Hero World Tour Definitive Edition for PC (D3D9)

### System information
- GPU: NVIDIA GeForce GTX 1060 6GB
- Driver: 520.56.06
- Wine version: wine-7.18 (Staging)
- DXVK version: v1.10.1-953-g53a0c372
- System version: Manjaro Linux x86_64 5.15.74

### Apitrace file(s)
- The game does not softlock without DXVK.

### Log files
- d3d9.log: [GHWT_Definitive_d3d9.log](https://github.com/doitsujin/dxvk/files/9956946/GHWT_Definitive_d3d9.log)

**Comment by Blisto91** [comment]
Sadness.

Could you check a quick random thing? Try making a file called `dxvk.conf` where the log file above appeared and insert `dxvk.enableGraphicsPipelineLibrary = False` and see if that changes anything? I'm not expecting it to, just to make sure.
To make sure it had picked up you can also enable the hud through there and verify it shows up `dxvk.hud = 1`.

**Comment by doitsujin** [comment]
> The game does not softlock without DXVK.

the whole point of an apitrace is that we can see what the game is doing and perhaps reproduce the bug by replaying a working trace on DXVK.

**Comment by ZedekThePlagueDoctor** [comment]
> the whole point of an apitrace is that we can see what the game is doing and perhaps reproduce the bug by replaying a working trace on DXVK.

I'll remember that for future reference, I'll get an apitrace if I can get anything to happen consistently! I've restarted my computer and so far it doesn't seem to be happening, I was able to play 5 songs with no softlocks of any sort. I've made sure that DXVK is enabled by enabling the HUD.

I'll provide further information or apitraces in the future if it happens again. Thanks!

**Comment by K0bin** [comment]
Please attach an apitrace, even if it does not softlock when you make that apitrace.

I don't own the game and it's impossible to buy a new copy with an unused license to make an online account.

**Comment by ZedekThePlagueDoctor** [comment]
I've found that trying to auto-launch into a venue causes a constant softlock. This does not happen with WineD3D, and seems to be DXVK specific.

This seems to happen with both v1.10.1-953-g53a0c372 and v1.10.1-988-gfddbcbde. Attached is a d3d9 log for v1.10.1-988-gfddbcbde, as well as an apitrace captured using WineD3D.

d3d9.log: [GHWT_Definitive_d3d9.log](https://github.com/doitsujin/dxvk/files/9984600/GHWT_Definitive_d3d9.log)
APITrace file (WineD3D): [MEGA](https://mega.nz/file/pOVyxRoJ#aTi4wHZcnt23nHmQrTt4npCEljC6xRNEqFM7zB5RaT4) [MediaFire](https://www.mediafire.com/file/5fij7auxoa32q85/GHWT_Definitive_autoboot.zip/file)

**Comment by K0bin** [comment]
Did you search for players in that apitrace?

It ends after this: 
![image](https://user-images.githubusercontent.com/1131720/201390776-936401c9-6894-45c3-9379-482f751dd7d0.png)

**Comment by ZedekThePlagueDoctor** [comment]
I've not searched for any players in that trace, this is just immediately booting into a song. Softlocking when searching for players or starting a song seems to be random and unpredictable, but doing this softlocks every time.

I recall letting the apitrace finish the actual intro sequence / loading (which the screenshot has not done), but I can take a fresh trace and verify that for sure.

**Comment by ZedekThePlagueDoctor** [comment]
I've taken a fresh WineD3D APITrace which is pretty generic, no fancy debug HUD or anything like that. This trace is a simple trace, only booting into the song and letting the highway rise.

I have tested the same using DXVK v1.10.1-988-gfddbcbde, and it seems to be working this time around, where it would ordinarily softlock. Attached is a DXVK log file, the APITrace, and video footage of where the trace _should_ end at.

If this does not cause a softlock, I can try to grab an extended APITrace online which includes searching for players.

![2022-11-11_19-15](https://user-images.githubusercontent.com/13931826/201446706-799220d1-73cc-494c-972f-cdc282684a76.png)

APITrace file (WineD3D): https://www.mediafire.com/file/ycpwlg8i1ibsb5m/GHWT_Definitive.trace.zip/file
Video: https://streamable.com/oyckls
d3d9.log: [GHWT_Definitive_d3d9.log](https://github.com/doitsujin/dxvk/files/9993719/GHWT_Definitive_d3d9.log)

**Comment by K0bin** [comment]
Regular songs work fine. I think it only happens in online play.

**Comment by ZedekThePlagueDoctor** [comment]
Do you want an online play APItrace that involves playing through several songs?

**Comment by K0bin** [comment]
Yes.

**Comment by ZedekThePlagueDoctor** [comment]
I'm going to try to get the trace uploaded, it's fairly large (18.9 GB zipped), but I've taken a smaller one in the meantime. The pending trace is an online trace with a test of 3 or 4 songs. This is another auto-boot WineD3D trace, which seems to constantly softlock me if I try to do the same using DXVK. I have tried this using DXVK v1.10.1-988-gfddbcbde, and DXVK 2.0, with the same results.

Does this cause a softlock for you? DXVK freezes at the image below, this is very early during loading before most assets are visually loaded.

![2022-11-11_21-10](https://user-images.githubusercontent.com/13931826/201453878-3a85a7df-d052-430e-9530-9e78763475a9.png)

APITrace file (WineD3D): https://www.mediafire.com/file/sarccik2rjoafto/GHWT_Definitive_autoboot.trace

**Comment by K0bin** [comment]
That trace finishes just fine too. If you don't need to play online, can you list the **precise** steps I'd have to take to make it softlock?

**Comment by ZedekThePlagueDoctor** [comment]
I should be able to give instructions, are you trying to run the game and perform these steps yourself?

In the meantime, I've uploaded a lengthy APITrace which involves playing online through a handful of songs. This was taken with WineD3D again and may or may not cause a softlock when played through DXVK.

APITrace file (WineD3D): https://www.mediafire.com/file/qmbx6whl7pmqsen/GHWT_Definitive_online.zip/file

**Comment by ZedekThePlagueDoctor** [comment]
Using DXVK version 1.10.3 does not softlock, but likely does not have the memory management fixes required to "properly" play GHWT. Currently, I can launch straight into a song using v2.0 and experience a softlock immediately, but v1.10.3 does not do this and continues fine.

With 2.0, I have tried wine versions `7.20 staging`, `lutris-7.2-2-x86_64`, and `lutris-6.13-3-x86_64` in an attempt to debug the issue. I have also tried this with both the Plasma and XFCE desktop environments to see if it is somehow a DE / window manager issue.

So far, it seems to be an issue isolated to DXVK specifically.

**Comment by K0bin** [comment]
Yes, I get it, it's a DXVK bug. I still cannot reproduce it. Single player songs work fine for me and I cannot access multiplayer.

**Comment by SeongGino** [comment]
> I've found that trying to auto-launch into a venue causes a constant softlock.

I know this is an old thread, but it is an ongoing one and thought to try and reproduce it myself.

At least, as of:

- Wine-GE 8-26
- DXVK 2.3.1
- Plasma 5.27.10
- NVIDIA 550.67 (w/ a 3060ti)
- GHWT:DE Launcher 2.2.2
- GHWT:DE v1.3.0.7 (`11/15/2023` in-game)

The game does **not** softlock for me when auto-starting a song, and the game played through it the whole way. Granted, it *did* technically softlock after the given song finished (specifically after confirming/dismissing the performance report stuff), but I'm not sure if this is a bug or a "feature" of the mod.

Unfortunately I can't comment on online stability or through multiple single player QP's/campaign yet (waiting to make an adapter for my Wiitar atm), but given that Auto Start was the most consistent reproduction method, perhaps this was fixed at some point? ~~I will dread if this is still a concern though, as apitracing through a small portion of GH3 (for a bug that wasn't a DXVK bug, but the game itself, lol) was like secondhand torture performance-wise, so I don't look forward to having to do that for potentially *several minutes* of gameplay, aha.~~

**Comment by Blisto91** [comment]
@ZedekThePlagueDoctor @SeongGino Could you two give this another shot with dxvk 2.5.1

**Comment by Blisto91** [comment]
@ZedekThePlagueDoctor @SeongGino Frinedly ping. Try with latest dxvk if you can
